skimage/__init__.py

```diff
@@ -159,3 +160,7 @@ def _raise_build_error(e):
 
 
 del warnings, functools, osp, imp, sys
+
+
+def lookfor(search_text):
+    return np.lookfor(search_text, sys.modules[__name__])
```

Comment:
At the top I would do something like:
```python
from numpy import lookfor as _lookfor
```
and then later, somewhere before the `del` statement do
```python
_module = sys.modules['name']
```
The function body would then be
```python
    return _lookfor(search_text, _module)
```

This would address two issues
-  `np` will not end up in the skimage namespace
-  `sys.modules` isn't available for use in `lookfor` because it was removed by `del` above